dvc/template.py

```diff
@@ -0,0 +1,185 @@
+            .replace("DVC_METRIC_DATA", "")
+            .replace(",", "")
+        )
+
+    @staticmethod
+    def fill(template_path, data, priority_datafile=None):
+        with open(template_path, "r") as fobj:
+            result_content = fobj.read()
+
+        for placeholder in Template.get_data_placeholders(template_path):
+            file = Template.get_datafile(placeholder)
+
+            if not file or priority_datafile:
+                key = priority_datafile
+            else:
+                key = file
+
+            result_content = result_content.replace(
+                placeholder,
+                json.dumps(
+                    data[key],
+                    indent=Template.INDENT,
+                    separators=Template.SEPARATORS,
+                    sort_keys=True,
+                ),
+            )
+
+        return result_content
+
+
+class DefaultLinearTemplate(Template):
+    TEMPLATE_NAME = "default"
+
+    DEFAULT_CONTENT = {
+        "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
+        "data": {"values": Template.METRIC_DATA_STRING},
+        "mark": {"type": "line"},
+        "encoding": {
+            "x": {"field": "x", "type": "quantitative"},
+            "y": {"field": "y", "type": "quantitative"},
+            "color": {"field": "rev", "type": "nominal"},
+        },
+    }
+
+
+class DefaultConfusionTemplate(Template):
+    TEMPLATE_NAME = "confusion_matrix"
+    DEFAULT_CONTENT = {
+        "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
+        "data": {"values": Template.METRIC_DATA_STRING},
+        "mark": "rect",
+        "encoding": {
+            "x": {
+                "field": "predicted",
+                "type": "nominal",
+                "sort": "ascending",
+            },
+            "y": {"field": "actual", "type": "nominal", "sort": "ascending"},
+            "color": {"aggregate": "count", "type": "quantitative"},
+            "facet": {"field": "rev", "type": "nominal"},
+        },
+    }
+
+
+class PlotTemplates:
+    TEMPLATES_DIR = "plot"
+    TEMPLATES = [DefaultLinearTemplate, DefaultConfusionTemplate]
+
+    @cached_property
+    def templates_dir(self):
+        return os.path.join(self.dvc_dir, self.TEMPLATES_DIR)
+
+    @cached_property
+    def default_template(self):
+        return os.path.join(self.templates_dir, "default.json")
+
+    def get_template(self, path):
+        t_path = os.path.join(self.templates_dir, path)
+        if os.path.exists(t_path):
+            return t_path
+
+        regex = re.compile(re.escape(t_path) + ".*")
+        for root, _, files in os.walk(self.templates_dir):
+            for file in files:
+                full_file = os.path.join(root, file)
+                if regex.findall(full_file):
+                    return full_file
+
+        raise TemplateNotFound(path)
+
+    def __init__(self, dvc_dir):
+        self.dvc_dir = dvc_dir
+
+        if not os.path.exists(self.templates_dir):
```

Comment:
do we gitignore it? 

does it mean that every time I run a new version (any command) of DVC I'll see this new dir created?

Reply:
We should not `gitignore` it, user might want to store his own plots here to not clutter his main repo.

Alternative will be to gitignore it and check whether this directory exists each time `plot` is used and initialize if not. 

I think we should go with 1st approach